File: Automated Test Machine.py
import random
import time
import os

# inputs
username = ["Kimmy", "Kim", "Kimberly"]
passcode = ["Kimmy123","Kim123", "Kimberly123"]
bank_options = ["Check Balance", "Withdraw", "Deposit", "Log to another Account", "Quit"]
Balance = [1000, 7000, 15000]  # Initial balance, adjust as needed

def clear_screen():
    if os.name == 'nt':
        os.system('cls')
    else:
        os.system('clear')

def welcomemessage():
    print(f'\n-----------------------------------\nWelcome to BEL Banking!\n-----------------------------------')

def login():
    user_input = input("Enter your username: ")
    pass_input = input("Enter your password: ")
    access_check(user_input, pass_input)

def access_check(user_input, pass_input):
    if user_input in username and pass_input == passcode[username.index(user_input)]:
        print(f'Access Granted for {user_input}!')
        print('Entering account', end="")
        for dot in range(3):
            time.sleep(1)
            print(".", end="", flush = True)
        presentchoices(user_input)
    else:
        print(f'Access Denied!')
        login()

def presentchoices(user_input):
    i = 1
    print(f'\nChoose only one from the following options:')
    for option in bank_options:
        print(f'[{i}]  {option}')
        i+= 1
    print(f'-----------------------------------')
    get_userchoice(user_input)

def get_userchoice(user_input):
    try:
        user_action = int(input("Choose an action: ").strip().title())
        if user_action == 1:
            checkbalance(user_input)
        elif user_action == 2:
            withdraw(user_input)
        elif user_action == 3:
            deposit(user_input)
        elif user_action == 4:
            login()
        elif user_action == 5:
            quit()
        else:
            print("Invalid choice.")
            get_userchoice(user_input)

    except ValueError:
        print("Please enter a number.")
        get_userchoice(user_input)

def checkbalance(user_input):
    user_index = username.index(user_input)
    print(f"Your current balance is: ${Balance[user_index]}.")
    presentchoices(user_input)

def withdraw(user_input):
    user_index = username.index(user_input)
    withdraw_amt = int(input("Enter amount to be withdrawn: "))
    if Balance[user_index] >= withdraw_amt:
        Balance[user_index] -= withdraw_amt
        print(f'Withdrawing {withdraw_amt}', end="")
        for dot in range(3):
            time.sleep(1)
            print(".", end="", flush = True)
        print(f"${withdraw_amt} has been successfully withdrawn from your account!")
    else:
        print(f"Insufficient balance. You cannot withdraw {withdraw_amt}.")
    presentchoices(user_input)

def deposit(user_input):
    user_index = username.index(user_input)
    deposit_amt = int(input("Enter amount to be deposited: "))
    Balance[user_index] += deposit_amt
    print(f'Depositing {deposit_amt}', end="")
    for dot in range(3):
        time.sleep(1)
        print(".", end="", flush=True)
    print(f"${deposit_amt} has been successfully deposited to your account!")
    presentchoices(user_input)

def quit():
    print(f"Thank you for banking with us!")
    exit()

welcomemessage()
login()

File: Automated Test Machine.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
#include <unistd.h>
#include <algorithm>

std::vector<std::string> username = {"Kimmy", "Kim", "Kimberly"};
std::vector<std::string> passcode = {"Kimmy123", "Kim123", "Kimberly123"};
std::vector<std::string> bank_options = {"Check Balance", "Withdraw", "Deposit", "Log to another Account", "Quit"};
std::vector<int> Balance = {1000, 7000, 15000}; // Initial balance, adjust as needed

void clear_screen() {
    if (system("clear") || system("cls")) {} // Clear the screen
}

void welcomemessage();

void login();

void access_check(const std::string& user_input, const std::string& pass_input);

void presentchoices(const std::string& user_input);

void get_userchoice(const std::string& user_input);

void checkbalance(const std::string& user_input);

void withdraw(const std::string& user_input);

void deposit(const std::string& user_input);

void quit();

int main() {
    welcomemessage();
    login();
    return 0;
}

void welcomemessage() {
    std::cout << "\n-----------------------------------\nWelcome to BEL Banking!\n-----------------------------------" << std::endl;
}

void login() {
    std::string user_input, pass_input;
    std::cout << "Enter your username: ";
    std::getline(std::cin, user_input);
    std::cout << "Enter your password: ";
    std::getline(std::cin, pass_input);
    access_check(user_input, pass_input);
}

void access_check(const std::string& user_input, const std::string& pass_input) {
    auto it = std::find(username.begin(), username.end(), user_input);
    if (it != username.end() && pass_input == passcode[std::distance(username.begin(), it)]) {
        std::cout << "Access Granted for " << user_input << "!" << std::endl;
        std::cout << "Entering account";
        for (int dot = 0; dot < 3; dot++) {
            usleep(1000000); // Sleep for 1 second
            std::cout << ".";
            std::cout.flush();
        }
        std::cout << std::endl;
        presentchoices(user_input);
    } else {
        std::cout << "Access Denied!" << std::endl;
        login();
    }
}

void presentchoices(const std::string& user_input) {
    std::cout << "\nChoose only one from the following options:" << std::endl;
    for (size_t i = 0; i < bank_options.size(); i++) {
        std::cout << "[" << i + 1 << "] " << bank_options[i] << std::endl;
    }
    std::cout << "-----------------------------------" << std::endl;
    get_userchoice(user_input);
}

void get_userchoice(const std::string& user_input) {
    try {
        int user_action;
        std::cout << "Choose an action: ";
        std::cin >> user_action;
        std::cin.ignore(); // Clear input buffer
        if (user_action >= 1 && user_action <= bank_options.size()) {
            switch (user_action) {
                case 1:
                    checkbalance(user_input);
                    break;
                case 2:
                    withdraw(user_input);
                    break;
                case 3:
                    deposit(user_input);
                    break;
                case 4:
                    login();
                    break;
                case 5:
                    quit();
                    break;
            }
        } else {
            std::cout << "Invalid choice." << std::endl;
            get_userchoice(user_input);
        }
    } catch (...) {
        std::cout << "Please enter a number." << std::endl;
        get_userchoice(user_input);
    }
}

void checkbalance(const std::string& user_input) {
    auto it = std::find(username.begin(), username.end(), user_input);
    if (it != username.end()) {
        int user_index = std::distance(username.begin(), it);
        std::cout << "Your current balance is: $" << Balance[user_index] << std::endl;
        presentchoices(user_input);
    }
}

void withdraw(const std::string& user_input) {
    auto it = std::find(username.begin(), username.end(), user_input);
    if (it != username.end()) {
        int user_index = std::distance(username.begin(), it);
        int withdraw_amt;
        std::cout << "Enter amount to be withdrawn: ";
        std::cin >> withdraw_amt;
        if (Balance[user_index] >= withdraw_amt) {
            Balance[user_index] -= withdraw_amt;
            std::cout << "Withdrawing " << withdraw_amt;
            for (int dot = 0; dot < 3; dot++) {
                usleep(1000000); // Sleep for 1 second
                std::cout << ".";
                std::cout.flush();
            }
            std::cout << "$" << withdraw_amt << " has been successfully withdrawn from your account!" << std::endl;
        } else {
            std::cout << "Insufficient balance. You cannot withdraw " << withdraw_amt << "." << std::endl;
        }
        presentchoices(user_input);
    }
}

void deposit(const std::string& user_input) {
    auto it = std::find(username.begin(), username.end(), user_input);
    if (it != username.end()) {
        int user_index = std::distance(username.begin(), it);
        int deposit_amt;
        std::cout << "Enter amount to be deposited: ";
        std::cin >> deposit_amt;
        Balance[user_index] += deposit_amt;
        std::cout << "Depositing " << deposit_amt;
        for (int dot = 0; dot < 3; dot++) {
            usleep(1000000); // Sleep for 1 second
            std::cout << ".";
            std::cout.flush();
        }
        std::cout << "$" << deposit_amt << " has been successfully deposited to your account!" << std::endl;
        presentchoices(user_input);
    }
}

void quit() {
    std::cout << "Thank you for banking with us!" << std::endl;
    exit(0);
}
